cookbooks/bcpc-hadoop/recipes/hdfs_group_directories.rb

```diff
@@ -1,8 +1,15 @@
 Chef::Resource::RubyBlock.send(:include, Bcpc_Hadoop::Helper)
+Chef::Resource::RubyBlock.send(:include, Chef::Mixin::DeepMerge)
 
 ruby_block "hdfs_group_directories" do
   block do
-    # create the /groups dirs: mode, dirs, home, perms
-    dir_creation('groups', cluster_groups, '/groups', '0770')
+
+    # Create directories for existing LDAP groups.
+    dirinfo = node.default['bcpc']['hadoop']['hdfs']['groups']['dirinfo']
+    (cluster_groups).each do |group|
+      dirinfo[group] = deep_merge(dirinfo[group], { group: group })
```

Comment:
As mentioned above, please clarify that `group` will always be set to what is passed in the `name` of the resource.